src/services/event.js

```diff
@@ -5,98 +5,124 @@
+ * @property {string} EndDate The datetime, as a string, when the occurence ends
+ * @property {string} Location The location of the occurence
  */
 
 /**
  * @global
  * @typedef Event
- * @property {String} Event_ID
- * @property {String} Event_Name
- * @property {String} Event_Title
- * @property {String} Event_Type_Name
- * @property {Number} Category_Id
- * @property {String} Description
- * @property {EventOccurrence[]} Occurrences
+ * @property {String} Event_ID The ID of the event, from 25Live
+ * @property {String} Event_Name The internal name of the event
+ * @property {String} Event_Title The title, to be displayed, of the event
+ * @property {boolean} HasCLAWCredit Whether the event offers CL&W Credit
+ * @property {String} Description The description of the event
+ * @property {EventOccurrence[]} Occurrences All scheduled occurrences of the event
  */
 
 /**
- * Get all events
- * @return {Promise.<Event[]>} returns all the events
+ * @global
+ * @typedef AttendedEvent
+ * @property {String} CHDate The date and time that the user received CL&W credit
+ * @property {String} CHTermCD Term code of the event
+ * @property {String} Description Given description of the event
+ * @property {String} Event_Name The generic name of the event
+ * @property {String} Event_Title Specific title of the event
+ * @property {EventOccurrence[]} Occurrences All scheduled occurrences of the event
+ * @property {String} Organization Organization hosting the event
+ * @property {Number} Required Required CL&W credits for the user
  */
 
-const getAllEvents = () => http.get('events/25Live/All');
+/**
+ * Gets all events from the backend, and then formats and sorts them
+ * @returns {Event[]} All events
+ */
+const getAllEvents = async () => {
+  const allEvents = await http.get('events/25Live/All');
+  return allEvents.map((e) => formatevent(e)).sort(sortEventsByTime);
+};
 
-const getAllCLAWEvents = () => http.get('/events/25Live/CLAW');
+/**
+ * Gets upcoming CL&W events and formats them for display
+ * TODO: Unused. Consider removing
+ * @returns {Event[]} upcoming CL&W events
+ */
+const getCLWEvents = async () => {
+  const allEvents = await http.get('events/25Live/CLAW');
+  const now = Date.now();
+  return allEvents
+    .filter((e) => new Date(e.Occurrences[0].StartDate).getTime() > now)
+    .map((e) => formatevent(e))
+    .sort(sortEventsByTime);
+};
 
-const getAllGuestEvents = () => http.get('events/25Live/Public');
+/**
+ * Gets all public events from the backend, and then formats and sorts them
+ * @returns {Event[]} All events
+ */
+const getAllGuestEvents = async () => {
+  const allGuest = await http.get('events/25Live/Public');
+  return allGuest.map((e) => formatevent(e)).sort(sortEventsByTime);
+};
+
+/**
+ * Get chapel events attended by the user during the current term
+ * @return {AttendedEvent[]} all CL&W events attended by the user, formatted and sorted
+ */
+const getAttendedChapelEvents = async () => {
```

Comment:
If you're asking why I moved it from user.js to event.js, the simple explanation is that this keeps all event logic in the one service. Previously, formatevent had to be exposed outside the service so that user.js could reuse it, and the logic for sorting events was repeated in both services. This way, the event logic is better encapsulated, and any components or services  that  need to use events in some way, as user.js does to calculate CL&W credits, can use the event service and not have to repeat the logic.